iceoryx2-bb/cxx/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2024 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache Software License 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+# which is available at https://opensource.org/licenses/MIT.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+cmake_minimum_required(VERSION 3.22)
+
+project(iceoryx2-bb VERSION ${IOX2_VERSION_STRING} LANGUAGES CXX)
+
+set(PREFIX iceoryx2/v${CMAKE_PROJECT_VERSION})
+
+# include only lib -> includes are installed only once despite being used in static lib as well as shared lib
```

Comment:
I think this one can be removed. The following code crates the final library directly and I guess this will also remain a header only lib.

We used this trick in the C++ bindings in order to create static as well as shared libraries with the same build and did not want to add the header directly to one of the binary targets but have a third, header only lib, which the binary targets can depend on.